lib/std/crypto/25519/ed25519.zig

```diff
@@ -298,7 +298,9 @@ pub const Ed25519 = struct {
             if (std.debug.runtime_safety) {
                 const pk_p = try Curve.fromBytes(secret_key.publicKeyBytes());
                 const recomputed_kp = try generateDeterministic(secret_key.seed());
-                debug.assert(mem.eql(u8, &recomputed_kp.public_key.toBytes(), &pk_p.toBytes()));
+                if (!mem.eql(u8, &recomputed_kp.public_key.toBytes(), &pk_p.toBytes())) {
+                    return error.NonCanonical;
```

Comment:
Nit: `NonCanonicalKey`/`NonCanonicalSecretKey`? Just `NonCanonical` seems a bit nebulous.

Reply:
`NonCanonical` is shared with other cryptographic functions. Keeping the `crypto.Error` error set small helps maintain API consistency and mitigates the implications of exposing errors in public interfaces. At the function level, the only parameter is the secret key, and more detailed errors would not be particularly useful to applications.